modules/engine/src/index.ts

```diff
@@ -1525,6 +1529,96 @@ export class VectorEngine implements IVectorEngine {
     }
   }
 
+  private async runAuction(
+    params: EngineParams.RunAuction,
+  ): Promise<Result<ChannelRpcMethodsResponsesMap[typeof ChannelRpcMethods.chan_runAuction], EngineError>> {
+    const validate = ajv.compile(EngineParams.RunAuctionSchema);
+    const valid = validate(params);
+    if (!valid) {
+      return Result.fail(
+        new RpcError(RpcError.reasons.InvalidParams, "", this.publicIdentifier, {
+          invalidParamsError: validate.errors?.map((e) => e.message).join(","),
+          invalidParams: params,
+        }),
+      );
+    }
+
+    const payload: RunAuctionPayload = {
+      amount: params.amount,
+      senderPublicIdentifier: this.publicIdentifier,
+      senderAssetId: params.assetId,
+      senderChainId: params.chainId,
+      receiverPublicIdentifier: params.recipient,
+      receiverAssetId: params.recipientAssetId,
+      receiverChainId: params.recipientChainId,
+    };
+    this.evts[EngineEvents.RUN_AUCTION_EVENT].post(payload);
+
+    const inbox = getRandomBytes32();
```

Comment:
to be changed. Couldn't find a service to generate a random inbox with the format "_INBOX.xxxxxxx"

Reply:
changed